### ISSUE TITLE
Error handling does not work in setup_pytorch_env.bat 

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#65351]
For example, have a look at https://github.com/pytorch/pytorch/runs/3604309455  or  https://github.com/pytorch/pytorch/runs/3641805622

Source of the problems are those two lines
https://github.com/pytorch/pytorch/blob/88032d894311e5c0aed8bbc21a4306bc6be4af82/.jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat#L23-L24
In https://github.com/pytorch/pytorch/runs/3604309455#step:11:175 one can clearly see, that `conda install` command failed, but looks like error handling got pre-computed even before the script is run:
```
call conda install -y -q python=3.8 numpy mkl cffi pyyaml boto3 protobuf numba scipy=1.6.3 typing_extensions dataclasses libuv  
PackagesNotFoundError: The following packages are not available from current channels:

  - scipy=1.6.3
 if 0 NEQ 0 (exit /b 0  )  
```

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @ezyang @seemethere @malfet @pytorch/pytorch-dev-infra

**Comment by seemethere** [comment]
We should just move this script from batch in general.

Perhaps this should be re-written in bash?

**Comment by peterjc123** [comment]
@malfet @seemethere I have a PR for this. https://github.com/pytorch/pytorch/pull/57331 But there is no further progress there.

**Comment by suo** [comment]
Adding some more examples from this morning:
- https://github.com/pytorch/pytorch/runs/3660500680
- https://github.com/pytorch/pytorch/runs/3659483265

**Comment by malfet** [comment]
> * https://github.com/pytorch/pytorch/runs/3660500680
> * https://github.com/pytorch/pytorch/runs/3659483265


@zhouzhuojie can we check if miniconda install is cached in squid proxy? Or are Windows runners not using it?

**Comment by malfet** [comment]
Were fixed by  #57331, closing
